cli/src/main/java/com/box/l10n/mojito/cli/command/checks/GlossaryTermTrie.java

```diff
@@ -0,0 +1,92 @@
+package com.box.l10n.mojito.cli.command.checks;
+
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class GlossaryTermTrie {
+
+    private final Node rootNode = new Node();
+
+    public GlossaryTermTrie(List<GlossaryTerm> terms) {
+        for (GlossaryTerm term : terms) {
+            this.add(term);
+        }
+    }
+
+    public void add(GlossaryTerm term) {
+        List<String> words = CheckerUtils.getWordsInString(term.getTerm());
+        Node currentNode = rootNode;
+        for (String word : words) {
+            word = word.toLowerCase();
+            if (!currentNode.children.containsKey(word)) {
+                currentNode.children.put(word, new Node());
+            }
+            currentNode = currentNode.children.get(word);
+        }
+        currentNode.glossaryTerms.add(term);
+    }
+
+    public GlossaryCaseCheckerSearchResult runGlossaryCheck(String source) {
```

Comment:
probably want to mention in the class name (and related classes) and function name that it is a trie for the "case checking" too?